src/Components/Scenes/Hydration/JobDesigner/DiagramApp/SourceNode/SourceNodeWidget.js

```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+import color from '@edma/design-tokens/js/color';
+import { PortWidget } from '@projectstorm/react-diagrams';
+
+export default class SourceNodeWidget extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {};
+  }
+
+  render() {
+    const { name, size } = this.props.node;
+    const height = size * 0.5;
+    return (
+      <div>
+        <div
+          className={'source-node'}
+          style={{
+            position: 'relative',
+            width: size,
+            height,
+            backgroundColor: color.white,
```

Comment:
Would this (and colors in any other places) eventually need dark mode support, like with a ternary?

Reply:
Yes! I know MUI has several different approaches for doing theming. I personally prefer the idea of having explicit light and dark palettes which are set once on a toggle event and plugged into the theme provider: https://material-ui.com/customization/palette/#type-light-dark-theme. I think it's a lot cleaner than having light/dark ternary checks all over the code, so I decided to leave it ambiguous for now until that plan is more fleshed out.